### DISCUSSION TITLE
Issue related to the Virtual User count in the overview of a test run page

### DISCUSSION THREAD

**Discussion by anita-achari** [ludeknovy/jtl-reporter#278] [General]
Hi,
After running a test with 20 users (Number of threads) using JMeter and uploading the JTL report to JTL- reporter, overview of the test run shows the number of virtual users as 1. I have attached the screenshot here.

<img width="827" alt="image" src="https://github.com/ludeknovy/jtl-reporter/assets/85698769/7208a487-2b7f-4c12-82bb-63015c7be16f">

**Comment by ludeknovy** [comment]
Hi @anita-achari - Do you use multiple machines to generate the load (distributed testing https://jmeter.apache.org/usermanual/jmeter_distributed_testing_step_by_step.html) ?

**Reply by anita-achari** [reply]
No. We are not using multiple machines.

**Reply by ludeknovy** [reply]
Ok. Could you please provide an anonymized sample of the data?

**Reply by anita-achari** [reply]
Sample data of the JTL report : 

timeStamp,elapsed,label,responseCode,responseMessage,threadName,dataType,success,failureMessage,bytes,sentBytes,grpThreads,allThreads,URL,Filename,Latency,SampleCount,ErrorCount,Hostname,IdleTime,Connect
1715235552909,9920,TestCaseLabel,200,OK,TestCaseThreadName 1-1,text,true,,21108474,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,8060,1,0,PA-Machine,0,454
1715235583094,9134,TestCaseLabel,200,OK,TestCaseThreadName 1-2,text,true,,21108486,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,7383,1,0,PA-Machine,0,40
1715235614679,9690,TestCaseLabel,200,OK,TestCaseThreadName 1-3,text,true,,21108527,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,7863,1,0,PA-Machine,0,35
1715235646270,9803,TestCaseLabel,200,OK,TestCaseThreadName 1-4,text,true,,21108581,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,8068,1,0,PA-Machine,0,23
1715235677854,9547,TestCaseLabel,200,OK,TestCaseThreadName 1-5,text,true,,21108647,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,7730,1,0,PA-Machine,0,24
1715235709440,9893,TestCaseLabel,200,OK,TestCaseThreadName 1-6,text,true,,21108506,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,8031,1,0,PA-Machine,0,28
1715235741032,9029,TestCaseLabel,200,OK,TestCaseThreadName 1-7,text,true,,21108513,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,7245,1,0,PA-Machine,0,28
1715235772602,9479,TestCaseLabel,200,OK,TestCaseThreadName 1-8,text,true,,21109205,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,7609,1,0,PA-Machine,0,33
1715235804203,9168,TestCaseLabel,200,OK,TestCaseThreadName 1-9,text,true,,21109154,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,7358,1,0,PA-Machine,0,40
1715235835773,9532,TestCaseLabel,200,OK,TestCaseThreadName 1-10,text,true,,21108514,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,7718,1,0,PA-Machine,0,23
1715235867366,9142,TestCaseLabel,200,OK,TestCaseThreadName 1-11,text,true,,21108517,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,7402,1,0,PA-Machine,0,21
1715235898941,9807,TestCaseLabel,200,OK,TestCaseThreadName 1-12,text,true,,21108643,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,8049,1,0,PA-Machine,0,47
1715235930519,9489,TestCaseLabel,200,OK,TestCaseThreadName 1-13,text,true,,21109149,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,7740,1,0,PA-Machine,0,28
1715235962096,9389,TestCaseLabel,200,OK,TestCaseThreadName 1-14,text,true,,21108669,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,7323,1,0,PA-Machine,0,35
1715235993667,9438,TestCaseLabel,200,OK,TestCaseThreadName 1-15,text,true,,21108642,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,7712,1,0,PA-Machine,0,32
1715236025242,8892,TestCaseLabel,200,OK,TestCaseThreadName 1-16,text,true,,21109238,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,7147,1,0,PA-Machine,0,39
1715236056807,9971,TestCaseLabel,200,OK,TestCaseThreadName 1-17,text,true,,21108493,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,8222,1,0,PA-Machine,0,38
1715236088376,9778,TestCaseLabel,200,OK,TestCaseThreadName 1-18,text,true,,21109084,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,8017,1,0,PA-Machine,0,27
1715236119938,9422,TestCaseLabel,200,OK,TestCaseThreadName 1-19,text,true,,21108825,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,7585,1,0,PA-Machine,0,27
1715236151486,9810,TestCaseLabel,200,OK,TestCaseThreadName 1-20,text,true,,21108555,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,7964,1,0,PA-Machine,0,19

**Reply by ludeknovy** [reply]
The provided CSV sample contains only `all_threads` = 1, so if your full file looks like this, indeed, the virtual users will be 1.

I've edited your sample, and set a couple of last lines `all_threads` = 2. And the virtual users changed to 2, so it seems to be working correctly.

timeStamp,elapsed,label,responseCode,responseMessage,threadName,dataType,success,failureMessage,bytes,sentBytes,grpThreads,allThreads,URL,Filename,Latency,SampleCount,ErrorCount,Hostname,IdleTime,Connect
1715235552909,9920,TestCaseLabel,200,OK,TestCaseThreadName 1-1,text,true,,21108474,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,8060,1,0,PA-Machine,0,454
1715235583094,9134,TestCaseLabel,200,OK,TestCaseThreadName 1-2,text,true,,21108486,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,7383,1,0,PA-Machine,0,40
1715235614679,9690,TestCaseLabel,200,OK,TestCaseThreadName 1-3,text,true,,21108527,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,7863,1,0,PA-Machine,0,35
1715235646270,9803,TestCaseLabel,200,OK,TestCaseThreadName 1-4,text,true,,21108581,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,8068,1,0,PA-Machine,0,23
1715235677854,9547,TestCaseLabel,200,OK,TestCaseThreadName 1-5,text,true,,21108647,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,7730,1,0,PA-Machine,0,24
1715235709440,9893,TestCaseLabel,200,OK,TestCaseThreadName 1-6,text,true,,21108506,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,8031,1,0,PA-Machine,0,28
1715235741032,9029,TestCaseLabel,200,OK,TestCaseThreadName 1-7,text,true,,21108513,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,7245,1,0,PA-Machine,0,28
1715235772602,9479,TestCaseLabel,200,OK,TestCaseThreadName 1-8,text,true,,21109205,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,7609,1,0,PA-Machine,0,33
1715235804203,9168,TestCaseLabel,200,OK,TestCaseThreadName 1-9,text,true,,21109154,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,7358,1,0,PA-Machine,0,40
1715235835773,9532,TestCaseLabel,200,OK,TestCaseThreadName 1-10,text,true,,21108514,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,7718,1,0,PA-Machine,0,23
1715235867366,9142,TestCaseLabel,200,OK,TestCaseThreadName 1-11,text,true,,21108517,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,7402,1,0,PA-Machine,0,21
1715235898941,9807,TestCaseLabel,200,OK,TestCaseThreadName 1-12,text,true,,21108643,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,8049,1,0,PA-Machine,0,47
1715235930519,9489,TestCaseLabel,200,OK,TestCaseThreadName 1-13,text,true,,21109149,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,7740,1,0,PA-Machine,0,28
1715235962096,9389,TestCaseLabel,200,OK,TestCaseThreadName 1-14,text,true,,21108669,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,7323,1,0,PA-Machine,0,35
1715235993667,9438,TestCaseLabel,200,OK,TestCaseThreadName 1-15,text,true,,21108642,2764,1,1,htpps://test-env.com/bl/TestManager.svc/TestManager,,7712,1,0,PA-Machine,0,32
1715236025242,8892,TestCaseLabel,200,OK,TestCaseThreadName 1-16,text,true,,21109238,2764,1,2,htpps://test-env.com/bl/TestManager.svc/TestManager,,7147,1,0,PA-Machine,0,39
1715236056807,9971,TestCaseLabel,200,OK,TestCaseThreadName 1-17,text,true,,21108493,2764,1,2,htpps://test-env.com/bl/TestManager.svc/TestManager,,8222,1,0,PA-Machine,0,38
1715236088376,9778,TestCaseLabel,200,OK,TestCaseThreadName 1-18,text,true,,21109084,2764,1,2,htpps://test-env.com/bl/TestManager.svc/TestManager,,8017,1,0,PA-Machine,0,27
1715236119938,9422,TestCaseLabel,200,OK,TestCaseThreadName 1-19,text,true,,21108825,2764,1,2,htpps://test-env.com/bl/TestManager.svc/TestManager,,7585,1,0,PA-Machine,0,27
1715236151486,9810,TestCaseLabel,200,OK,TestCaseThreadName 1-20,text,true,,21108555,2764,1,2,htpps://test-env.com/bl/TestManager.svc/TestManager,,7964,1,0,PA-Machine,0,19

**Reply by rama-ram** [reply]
Hi , 
First of all thanks for providing such a wonderful tool to the community. Offlate, I am facing a similar problem as reported here, 
1. I am running distributed Load test programmatically on multiple hosts
2. If I extract the csv before sending it to JTL reporter and generate an html out of it, I can see the VU reflecting correctly in graphs , there were 2 VU per hosts, and total 2 hosts, so 4 VU in total
<img width="963" alt="Screenshot 2025-04-17 at 5 53 59 PM" src="https://github.com/user-attachments/assets/1899b885-008e-4698-a409-d94b023e4d68" />
But JTL Reporter shows 2 as VU, however sample sizes and all other metrics are correct. 
As per your note in [here](https://jtlreporter.site/docs/integrations/jmeter) 
```If you run your tests in distributed mode you need to provide Hostname in csv output. You can do it by setting jmeter.save.saveservice.hostname=true in jmeter.properties```
I had also set the property. Can you please let me know what change is needed to reflect the correct VU
<img width="923" alt="Screenshot 2025-04-17 at 5 57 28 PM" src="https://github.com/user-attachments/assets/d57ea398-f5da-45d5-9570-b0bf393febb3" />
